src/mainwin.cc

```diff
@@ -1947,7 +1948,15 @@ void MainWindow::sendToOctoPrint()
 		userFileName = fileInfo.baseName() + "." + fileFormat.toLower();
 	}
 
-	exportFileByName(this->root_geom, exportFileFormat, exportFileName.toLocal8Bit().constData(), exportFileName.toUtf8());
+    ExportInfo exportInfo;
+    exportInfo.format = exportFileFormat;
+    exportInfo.name2open = exportFileName.toLocal8Bit().constData();
+    exportInfo.name2display = exportFileName.toUtf8();
+    auto fnameba = activeEditor->filepath.toLocal8Bit();
+    const char* fname = activeEditor->filepath.isEmpty() ? "" : fnameba;
+    exportInfo.sourceFileName = fname;
```

Comment:
Looks like this block is identical in 3 places, so it should go into a separate function. This could be one in a anonymous namespace like in https://github.com/openscad/openscad/blob/a51718b59b5b7033753a02942c1913c5a75764d8/src/expr.cc#L48-L51